util.go

```diff
@@ -22,8 +22,10 @@ type offsetInfo struct {
 	Metadata string
 }
 
+// if we're not asking for the newest and known offset is positive, return that
+// if we are asking for the newest, known offset is irrelevant
 func (i offsetInfo) NextOffset(fallback int64) int64 {
-	if i.Offset > -1 {
+	if fallback == -2 && i.Offset > -1 {
```

Comment:
Is that `-2` a magic number?

Reply:
Yes, so is -1. They're used by sarama client to denote oldest or newest messages.